### ISSUE TITLE
Unexpected behavior with DataLoader.batches_per_epoch

### ISSUE THREAD

**Issue by epwalsh** [allenai/allennlp#4393]
From the discussion in https://github.com/allenai/allennlp/issues/4362.

If `batches_per_epochs` is less than the total number of batches in the dataset, there is no guarantee that all unique instances will be seen during training.

**Comment by mateuszpieniak** [comment]
@epwalsh It seems like `batches_per_epoch` applies also to validation set. Is it expected behaviour?

**Comment by epwalsh** [comment]
That is expected. If you only want `batches_per_epoch` set for training (like most use-cases, I assume) you can use a separate `validation_data_loader`.
